fem/tmop.hpp

```diff
@@ -708,30 +708,59 @@ class DiscreteAdaptTC : public TargetConstructor
 protected:
    // Discrete target specification.
    // Data is owned, updated by UpdateTargetSpecification.
-   Vector tspec;             //eta(x)
+   int   ncomp;
+   int   sizeidx,  skewidx,  aspectratioidx,  orientationidx;
+   Vector tspec;              //eta(x)
    Vector tspec_sav;
    Vector tspec_perth;       //eta(x+h)
    Vector tspec_pert2h;      //eta(x+2*h)
    Vector tspec_pertmix;     //eta(x+h,y+h)
+   // The new order for these vectors is
+   // eta1(x+h),eta2(x+h)...etan(x+h),eta1(y+h),eta2(y+h)...etan(y+h).
+   // same for tspec_pert2h and tspec_pertmix.
 
    // Note: do not use the Nodes of this space as they may not be on the
    // positions corresponding to the values of tspec.
+#ifdef MFEM_USE_MPI
+   ParFiniteElementSpace *ptspec_fes;
+#endif
    const FiniteElementSpace *tspec_fes;
+   const FiniteElementSpace *tspec_fesv;
 
    // Evaluation of the discrete target specification on different meshes.
    // Owned.
    AdaptivityEvaluator *adapt_eval;
 
+   void SetSerialDiscreteTargetBase(GridFunction &tspec_);
+#ifdef MFEM_USE_MPI
+   void SetParDiscreteTargetBase(ParGridFunction &tspec_);
+#endif
+
 public:
    DiscreteAdaptTC(TargetType ttype)
       : TargetConstructor(ttype),
+        ncomp(0),
+        sizeidx(-1), skewidx(-1), aspectratioidx(-1), orientationidx(-1),
         tspec(), tspec_fes(NULL), adapt_eval(NULL) { }
 
-   virtual ~DiscreteAdaptTC() { delete adapt_eval; }
+   virtual ~DiscreteAdaptTC()
+   {
+      delete adapt_eval;
+      delete tspec_fes;
+      delete tspec_fesv;
+   }
 
-   virtual void SetSerialDiscreteTargetSpec(GridFunction &tspec_);
```

Comment:
Lets keep this for now to have existing codes compiling. Inside we can just call the TargetSize version.

Reply:
Same for the Par version.